client/my-sites/patterns/components/page-view-tracker.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect } from 'react';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSelector } from 'calypso/state';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import getUserSetting from 'calypso/state/selectors/get-user-setting';
+
+type PatternsPageViewTrackerProps = {
+	category: string;
+	searchTerm: string;
+};
+
+export function PatternsPageViewTracker( { category, searchTerm }: PatternsPageViewTrackerProps ) {
+	const isLoggedIn = useSelector( isUserLoggedIn );
+	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
+
+	useEffect( () => {
+		const commonProperties = {
+			is_logged_in: isLoggedIn,
+			user_is_dev_account: isDevAccount ? '1' : '0',
+		};
+
+		if ( searchTerm ) {
+			recordTracksEvent( 'calypso_pattern_library_search', {
+				...commonProperties,
+				search_term: searchTerm,
+			} );
+		}
+
+		if ( category ) {
```

Comment:
```suggestion
		} else if ( category ) {
```
I just came to realize that we should only track `calypso_pattern_library_search` **or** `calypso_pattern_library_search` instead of both when users perform a search. Will the search return results specific to the selected category? If so, then perhaps we can include `category` as a property for `calypso_pattern_library_search`?